flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/enumerator/KinesisStreamsSourceEnumerator.java

```diff
@@ -211,32 +301,33 @@ private void assignSplits(List<KinesisShardSplit> discoveredSplits, Throwable th
             throw new KinesisStreamsSourceException("Failed to list shards.", throwable);
         }
 
+        splitTracker.addSplits(discoveredSplits);
+        updateLastSeenShardId(discoveredSplits);
+
         if (context.registeredReaders().size() < context.currentParallelism()) {
             LOG.info(
                     "Insufficient registered readers, skipping assignment of discovered splits until all readers are registered. Required number of readers: {}, Registered readers: {}",
                     context.currentParallelism(),
                     context.registeredReaders().size());
-            unassignedSplits.addAll(discoveredSplits);
             return;
         }
 
+        assignSplits();
+    }
+
+    private void assignSplits() {
```

Comment:
Can we rename this to a name with more context to emphasize the difference between enumerator and source reader?

Reply:
This method is performing split assignment to readers, IMO name represents the action well.

Renamed method above to clearer represent meaning.